### PR TITLE
Fix login and onboarding flows

### PR DESCRIPTION
**Pull request by afzalsayed96**
## What does this PR do?

When you sign up you can notice a slight jitter in the UI with invalid state while redirecting to `/get-started` route.

https://user-images.githubusercontent.com/14029371/161838212-a8bb3e9b-105e-477e-a2ce-5e8e2ef79824.mov

Similarly when you navigate to a protected page such as `/event-types` without being logged in you can see a glimpse of the invalid state

https://user-images.githubusercontent.com/14029371/161838498-1297bca4-29cb-4ea2-86ac-a3ee6dbad941.mov

The root cause of this issue is how `useRedirectToLoginIfUnauthenticated` and `useRedirectToOnboardingIfNeeded` hooks are implemented. They both employ `useEffect` hook to redirect. However, `useEffect` is run after a render is completed. Hence, we see a glimpse of invalid state as the loading of session/me query finishes but `useEffect` is yet to run.

The solution is for unauth redirection is simply to check for `session` value and only then render protected UI. And similarly for onboarding user redirect, it's imperative to check both the result & loading state of `me` query before rendering anything.

With the above changes, we make sure that protected UI is only shown after consulting the appropriate loading states and the data itself and we get a clean jitter-free transition.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Unsure

### CODE REVIEW

File: apps/web/components/Shell.tsx [hunk 1]
```diff
@@ -84,11 +85,7 @@ function useRedirectToOnboardingIfNeeded() {
   const query = useMeQuery();
   const user = query.data;
 
-  const [isRedirectingToOnboarding, setRedirecting] = useState(false);
-
-  useEffect(() => {
-    user && setRedirecting(shouldShowOnboarding(user));
-  }, [router, user]);
+  const isRedirectingToOnboarding = user && shouldShowOnboarding(user);
```
Comment (afzalsayed96) [inline apps/web/components/Shell.tsx:88]: I believe state is unnecessary here as any change in `query` will cause a rerender and `isRedirectingToOnboarding` can be considered a derived or computed state from `query.data`

File: apps/web/components/Shell.tsx [hunk 2]
```diff
@@ -201,218 +198,222 @@ export default function Shell(props: {
...
+            </main>
+          </div>
         </div>
-      </div>
-    </>
-  );
+      </>
+    );
+
+  return null;
```
Comment (afzalsayed96) [inline apps/web/components/Shell.tsx:416]: We return null when there's no `session`. This occurs when `session=null` and `loading=false`. The `useEffect` which redirects to `/auth/login` is run after this render.

We can also choose to show loader in this case.

**Comment by denik1981** [comment]
@alannnc @zomars 
Please check #2337 which provides more tests for the login/logout flow and normalizes the usage of the user fixture to decouple tests from the seed data. This won't help to test code changes in this PR but it will help assure everything is still working

File: apps/web/components/Shell.tsx [hunk 3]
```diff
@@ -76,6 +76,7 @@ function useRedirectToLoginIfUnauthenticated() {
 
   return {
     loading: loading && !session,
+    session: session,
```
Comment (alishaz-polymath) [inline apps/web/components/Shell.tsx:79]:
```suggestion
    session,
```

File: apps/web/components/Shell.tsx [hunk 4]
```diff
@@ -201,218 +198,222 @@ export default function Shell(props: {
...
-            <div
+          <div className="flex w-0 flex-1 flex-col overflow-hidden">
+            <main
               className={classNames(
-                props.centered && "mx-auto md:max-w-5xl",
-                props.flexChildrenContainer && "flex flex-1 flex-col",
-                !props.large && "py-8"
+                "relative z-0 flex-1 overflow-y-auto focus:outline-none",
+                status === "authenticated" && "max-w-[1700px]",
+                props.flexChildrenContainer && "flex flex-col"
```
Comment (alishaz-polymath) [inline apps/web/components/Shell.tsx:313]: Why is this being modified?

**Review by alishaz-polymath — CHANGES_REQUESTED** [review]
Minor change requested, and additional information requested. It is an improvement on the current state.

**Review by joeauyeung — COMMENTED** [review]
Hey, great PR! The transitions are now smooth. Just building off of @alishaz-polymath's question about why some of the class names have changed.

**Review by zomars — APPROVED** [review]
Reverted unnecessary diffs. Other than that code looks good and tested. Thank you for your contribution 🙏

**Review by alishaz-polymath — APPROVED** [review]
Thank you for your  contribution. Tested, works! 🙏

**Comment by denik1981** [comment]
@afzalsayed96 Thank you for this contribution. Quite helpful!

**Pull request merged**